google/fuzzbench
# Add default end to end test config file for testing; prepare for config validation refactoring

## What is this PR?
This PR helps for later development, when we need to have fixed end-to-end testing environment. In short, we need to run a unique experiment with a specified/default configuration for our end-to-end test.

This PR helps with this and starts the refactoring of reading config files, `common/config_utils.py`. We will move the validation logic from `experiment/run_experiment.py` to `common/config_utils.py` in later PR.

## How to test?
We modify several files to work together and achieve that:
1. `Makefile` and `compose/e2e-test.yaml`: The end-to-end test will start with `test-run-expeirment`, which is no different with `run-experiment` excepting additional environment variable: `E2E_INTEGRATION_TEST = 1`.
2. `docker/fuzzbench/Dockerfile` and `config.yaml`: copy the user-specified configuration as default behavior (which is not for end-to-end testing).
3. `fuzzbench/run_experiment.py`: it will first use the `config.yaml`. However, if `E2E_INTEGRATION_TEST` is set, then we use the default config for end-to-end test. Also, if `config['end_to_end_test']=true`, then we enqueue a fake job into the queue with `id='e2e_test'`.
4. `fuzzbench/test_e2e_run.py`: we check that a job with `id='e2e_test'` should exist.
